force-app/main/default/classes/PMT_RefundService.cls

```diff
@@ -190,20 +224,103 @@ public inherited sharing class PMT_RefundService {
         return response;
     }
 
-    public void processDML() {
-        Savepoint sp = Database.setSavepoint();
-        try {
-            if (!this.refundRecords.isEmpty()) {
-                insert this.refundRecords;
+    public npe01__OppPayment__c buildRefundRecord(npe01__OppPayment__c originalPayment, Decimal refundAmount) {
+        refundAmount = (refundAmount == null) ? originalPayment.npe01__Payment_Amount__c : refundAmount; 
```

Comment:
I was going to propose a second method call for full refunds that doesn't take a refundAmount, but I think a description on this method should be enough. Having `null` represent a full refund just doesn't feel great to me.

Reply:
I will remove the `null` check here. In the Controller, `refundAmount` will always have a value. This was a leftover from the previous WI